--- bglibs-1.005/installer.c.orig	Wed Dec 18 00:26:02 2002
+++ bglibs-1.005/installer.c	Sun Jan 19 11:25:34 2003
@@ -29,6 +29,10 @@
 static void setmodes(const char* filename,
 		     unsigned uid, unsigned gid, unsigned mode)
 {
+  if (uid == (unsigned)-1)
+    uid = getuid();
+  if (gid == (unsigned)-1)
+    gid = getgid();
   if (chown(filename, uid, gid) == -1)
     diefsys("Could not set owner or group for", filename);
   if (chmod(filename, mode) == -1)
