--- bglibs-1.009/Makefile.orig	2003-04-07 19:47:25.000000000 +0200
+++ bglibs-1.009/Makefile	2003-06-01 11:57:28.000000000 +0200
@@ -126,6 +126,15 @@
 	) >compile
 	chmod 755 compile
 
+compile-no-o: compile
+	(  \
+	  echo '#!/bin/sh'; \
+	  echo 'source=$$1; shift'; \
+	  echo 'base=`echo "$$source" | sed -e s:\\\\.c$$::`'; \
+	  echo exec `head -1 conf-cc` -I.  '-c $$source $${1+"$$@"}'; \
+	) >compile-no-o
+	chmod 755 compile-no-o
+
 conf_home.c: conf-home
 	head -1 conf-home | \
 	  sed -e 's/"/\\"/g' \
@@ -679,7 +688,7 @@
 pwcmp/module.o: compile pwcmp/module.c pwcmp/module.h
 	./compile pwcmp/module.c
 
-selftests: selftests.sh
+selftests: selftests.sh compile-no-o
 	sh selftests.sh
 
 str/alloc.o: compile str/alloc.c str/str.h selftest.c iobuf/iobuf.h msg/msg.h str/str.h
--- bglibs-1.009/selftests.sh.orig	2003-04-07 19:47:25.000000000 +0200
+++ bglibs-1.009/selftests.sh	2003-06-01 11:56:44.000000000 +0200
@@ -14,7 +14,7 @@
   rm -f $t/*
   sed -e '1,/^#ifdef SELFTEST_EXP$/d' -e '/^#endif/,$d' $1 >$t/test.exp
 
-  ./compile $1 -DSELFTEST_MAIN -o $t/test.o || {
+  ./compile-no-o $1 -DSELFTEST_MAIN -o $t/test.o || {
     echo "=====> Compile failed! <====="
     return 1
   }
